serde-generate/runtime/typescript/serde/binaryDeserializer.ts

```diff
@@ -0,0 +1,113 @@
+import { BigNumber } from '@ethersproject/bignumber';
+import { Int64LE, Uint64LE } from 'int64-buffer';
+import { Deserializer } from './deserializer';
+import { Readable } from 'stream';
+
+export abstract class BinaryDeserializer implements Deserializer {
+  public static readonly MAX_VALUE = 2147483647;
+  public data: Readable;
+  private readonly totalLength: number;
+
+  private static makeReadable(data: Uint8Array): Readable {
+    const r = new Readable();
+    r.push(Buffer.from(data));
+    r.push(null);
+    return r;
+  }
+
+  protected constructor(data: Uint8Array) {
+    this.data = BinaryDeserializer.makeReadable(data);
+    this.totalLength = this.data.readableLength;
+  }
+
+  abstract deserializeLen(): number;
+
+  abstract deserializeVariantIndex(): number;
+
+  abstract checkThatKeySlicesAreIncreasing(
+      key1: [number, number],
+      key2: [number, number]
+  ): void;
+
+  public deserializeStr(): string {
+    const value = this.deserializeBytes();
+    return String.fromCharCode.apply(null, Array.from(value));
```

Comment:
Text encode/decode should probably go through [`TextEncoder/TextDecoder`](https://developer.mozilla.org/en-US/docs/Web/API/Encoding_API):
```
var uint8array = new TextEncoder("utf-8").encode("¢");
var string = new TextDecoder("utf-8").decode(uint8array);
```